torch/_dynamo/variables/builder.py

```diff
@@ -899,6 +899,21 @@ def wrap_tensor(self, value: torch.Tensor):
         # then the relevant SubgraphTracer will lift it to being an input of
         # the subgraph.
         # See NOTE [HigherOrderOperator tracing design] for more details.
+
+        if not self.tx.output.export:
+            # Export has (supposedly) valid cases for fake tensors as inputs here.
+            # I am not convinced, atm, but out of scope for what this assert was added for (protecting value checks
+            # in real_value_tensor_positive_aliases in the common case)
+            assert not isinstance(value, torch._subclasses.fake_tensor.FakeTensor)
+
+        if value in self.tx.output.real_value_tensor_positive_aliases:
+            stored_value = self.tx.output.real_value_tensor_positive_aliases[value]
+            # TODO(voz): Decently common pattern, refactor at some point.
+            dup_guard = self._make_dupe_guard(stored_value)
```

Comment:
This function has a TODO for handling globals. You are making this ping a lot more frequently, it seems like that should get fixed now?

Reply:
The TODO is a little more nuanced. It is not for globals, but for dupe guards that cross alias across a local and global. 

It's also overly restrictive, let me see if I can alleviate it a little.